dist/tools/endingcheck/check.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+#
+# Copyright (C) 2015 Kushal Singh <kushal.spiderman.singh@gmail.com>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+BRANCH=${1}
+FILEREGEX='\.([sScHh]|cpp)$'
```

Comment:
I would actually include all files, when I think about it, not just source code.